custom_components/tuya_local/devices/sobralik_water_fountain.yaml

```diff
@@ -0,0 +1,104 @@
+name: Sobralik Pet Water Fountain
+products:
+  - id: hpa87ady0hndivqn
+primary_entity:
+  entity: switch
+  icon: "mdi:water-pump"
+  dps:
+    - id: 1
+      type: boolean
+      name: switch
+    - id: 23
+      name: fault
+      type: bitfield
+      mapping:
+        - dps_val: 0
```

Comment:
What is the value for no fault? Normally that would be 0.  If these are bit numbers from the tuya documentation, then the values should be 1, 2 and 4.

Reply:
This query is blocking for merge

Reply:
I went ahead and assumed that this is a bitfield, the same as most such dps, and the "dps_val" in your change need to be translated to bit indexes. Rather than a fault attribute, I have created three different binary sensors, as these all seem quite important to trigger notifications etc.